mmv1/products/bigqueryanalyticshub/QueryTemplate.yaml

```diff
@@ -0,0 +1,132 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'QueryTemplate'
+description: |
+  Represents a BigQuery Query Template within a Data Exchange.
+  This resource defines a reusable SQL routine (e.g., a TVF) that can be
+  shared or executed via the Data Exchange.
+references:
+  guides:
+    'Official Documentation': 'https://cloud.google.com/bigquery/docs/analytics-hub-introduction'
+self_link: 'projects/{{project}}/locations/{{location}}/dataExchanges/{{data_exchange_id}}/queryTemplates/{{query_template_id}}'
+id_format: 'projects/{{project}}/locations/{{location}}/dataExchanges/{{data_exchange_id}}/queryTemplates/{{query_template_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/dataExchanges/{{data_exchange_id}}/queryTemplates?queryTemplateId={{query_template_id}}'
+base_url: 'projects/{{project}}/locations/{{location}}/dataExchanges/{{data_exchange_id}}/queryTemplates'
+update_verb: 'PATCH'
+update_mask: true
+import_format:
+  - 'projects/{{project}}/locations/{{location}}/dataExchanges/{{data_exchange_id}}/queryTemplates/{{query_template_id}}'
+timeouts:
+  insert_minutes: 20
+  update_minutes: 20
+  delete_minutes: 20
+examples:
+  - name: 'bigquery_analyticshub_querytemplate_basic'
+    primary_resource_id: 'querytemplate'
+    primary_resource_name: 'fmt.Sprintf("tf_test_my_data_exchange%s", context["random_suffix"]), fmt.Sprintf("tf_test_my_query_template%s", context["random_suffix"])'
+    region_override: 'us'
+    vars:
+      data_exchange_id: 'my_data_exchange'
+      query_template_id: 'qt1'
+      desc: 'example of query template'
+    ignore_read_extra:
+      - 'submit'
+custom_code:
+  post_update: 'templates/terraform/post_update/bigquery_analytics_hub_query_template.go.tmpl'
+  pre_update: 'templates/terraform/pre_update/bigquery_analytics_hub_query_template.go.tmpl'
+  post_create: 'templates/terraform/post_update/bigquery_analytics_hub_query_template.go.tmpl'
```

Comment:
Right now the fact that `post_update` logic is also run for `post_create` is not very visible from an ongoing maintenance perspective. I'd propose to put the shared logic in a function in `templates/.../constants` (https://googlecloudplatform.github.io/magic-modules/develop/custom-code/#add-reusable-variables-and-functions), then reference that from separate `post_update` and `post_create` template files.

Reply:
I see that in the latest commit you have separate `post_create` and `post_update` now, thank you! We may be able to reduce some duplicated code by extracting the common logic into a function in e.g. `constants/`. Could you explore that?